src/Api/SecretsManager/Controllers/SecretsController.cs

```diff
@@ -54,42 +62,101 @@ public async Task<SecretResponseModel> CreateAsync([FromRoute] Guid organization
         return new SecretResponseModel(result);
     }
 
     [HttpGet("secrets/{id}")]
     public async Task<SecretResponseModel> GetAsync([FromRoute] Guid id)
     {
         var secret = await _secretRepository.GetByIdAsync(id);
-        if (secret == null)
+        if (secret == null || !_currentContext.AccessSecretsManager(secret.OrganizationId))
+        {
+            throw new NotFoundException();
+        }
+
+        if (!await userHasReadAccessToSecret(secret))
         {
             throw new NotFoundException();
         }
+
         return new SecretResponseModel(secret);
     }
 
     [HttpGet("projects/{projectId}/secrets")]
     public async Task<SecretWithProjectsListResponseModel> GetSecretsByProjectAsync([FromRoute] Guid projectId)
     {
-        var secrets = await _secretRepository.GetManyByProjectIdAsync(projectId);
+        var project = await _projectRepository.GetByIdAsync(projectId);
+        if (project == null || !_currentContext.AccessSecretsManager(project.OrganizationId))
+        {
+            throw new NotFoundException();
+        }
+
+        var userId = _userService.GetProperUserId(User).Value;
+        var orgAdmin = await _currentContext.OrganizationAdmin(project.OrganizationId);
+        var accessClient = AccessClientHelper.ToAccessClient(_currentContext.ClientType, orgAdmin);
+
+        var secrets = await _secretRepository.GetManyByProjectIdAsync(projectId, userId, accessClient);
+
         var responses = secrets.Select(s => new SecretResponseModel(s));
         return new SecretWithProjectsListResponseModel(secrets);
     }
 
     [HttpPut("secrets/{id}")]
-    public async Task<SecretResponseModel> UpdateAsync([FromRoute] Guid id, [FromBody] SecretUpdateRequestModel updateRequest)
+    public async Task<SecretResponseModel> UpdateSecretAsync([FromRoute] Guid id, [FromBody] SecretUpdateRequestModel updateRequest)
     {
-        var result = await _updateSecretCommand.UpdateAsync(updateRequest.ToSecret(id));
+        var userId = _userService.GetProperUserId(User).Value;
+        var secret = updateRequest.ToSecret(id);
+        var result = await _updateSecretCommand.UpdateAsync(secret, userId);
         return new SecretResponseModel(result);
     }
 
-    // TODO Once permissions are setup for Secrets Manager need to enforce them on delete.
-    [HttpPost("secrets/delete")]
-    public async Task<ListResponseModel<BulkDeleteResponseModel>> BulkDeleteAsync([FromBody] List<Guid> ids)
+    [HttpPost("secrets/{organizationId}/delete")]
+    public async Task<ListResponseModel<BulkDeleteResponseModel>> BulkDeleteAsync([FromBody] List<Guid> ids, [FromRoute] Guid organizationId)
     {
-        var results = await _deleteSecretCommand.DeleteSecrets(ids);
+        var userId = _userService.GetProperUserId(User).Value;
+        var results = await _deleteSecretCommand.DeleteSecrets(ids, userId, organizationId);
```

Comment:
`[HttpPost("secrets/{organizationId}/delete")]`

I would recommend taking organization ID out of this. 

See `DeleteSecretCommand` comments on how we can remove needing to pass in organization ID.